src/vs/workbench/services/search/common/search.ts

```diff
@@ -433,9 +443,21 @@ export interface IRawSearchService {
 
 export interface IRawFileMatch {
 	base?: string;
+	/**
+	 * The path of the file relative to the containing `base` folder.
+	 * This path is exactly as it appears on the filesystem.
+	 */
 	relativePath: string;
```

Comment:
Do you think it's ok to basically replace relativePath with searchPath or do we actually need the real filesystem relativePath somewhere?

Reply:
Unfortunately, the `relativePath` is used in a couple of places where `searchPath` cannot (trivially) replace it:
1. In [rawSearchService.ts:139](https://github.com/microsoft/vscode/blob/1.42.1/src/vs/workbench/services/search/node/rawSearchService.ts#L139) it's used to construct the absolute path.
2. In [rawSearchService.ts:421 and :425](https://github.com/microsoft/vscode/blob/1.42.1/src/vs/workbench/services/search/node/rawSearchService.ts#L421-L425) it's probably used to display visual information about the file match to the user.

Usage in 1. is impossible to faithfully replicate with `searchPath`. Usage in 2. would (I suspect) change the visual feedback in the UI.

However, I did look at `IRawFileMatch` a before and I have a branch ready for another PR that removes the `size` field (it's not used anywhere): https://github.com/urbas/vscode/commit/01c314566758a7c18ec4a1abd74da83c410f000a

We could also remove the `basename`, which can easily be extracted from `relativePath`. I'll try to remove that together with `size` and open a separate pull request (after this one).